packages/react/src/utils.ts

```diff
@@ -127,14 +127,18 @@ export type RequiredKeysOf<BaseType> = Exclude<
  * Includes the data result object and a function for running the mutation.
  **/
 export type ActionHookResult<Data = any, Variables extends AnyVariables = AnyVariables> = RequiredKeysOf<Variables> extends never
-  ? [
-      ActionHookState<Data, Variables>,
-      (variables?: Variables, context?: Partial<OperationContext>) => Promise<ActionHookState<Data, Variables>>
-    ]
-  : [
-      ActionHookState<Data, Variables>,
-      (variables: Variables, context?: Partial<OperationContext>) => Promise<ActionHookState<Data, Variables>>
-    ];
+  ? ActionHookResultWithOptionalCallbackVariables<Data, Variables>
+  : ActionHookResultWithRequiredCallbackVariables<Data, Variables>;
+
```

Comment:
Note that I tried doing a ternary chain here, but this causes a lot of type errors